### PR TITLE
Meta parsers misc fix suggested in #1137

### LINKED ISSUE #1158: reflect the comments about yaml meta parsrr
I got comments from @b4n about my PR #1137.
There is no way to reopen PR. So I register my task here.

### PR DESCRIPTION
**Pull request by masatake**
Close #1158.

### CODE REVIEW

File: parsers/meta-m4.h [hunk 1]
```diff
@@ -28,8 +28,8 @@ struct m4ParserClient
 	langType lang;
 	void *data;
 
-	char quoteOpen;
-	char quoteClose;
+	const char quoteOpen;
+	const char quoteClose;
```
Comment (b4n) [inline parsers/meta-m4.h]: why?  how will `changequote()` be handled then?

File: parsers/yaml.c [hunk 2]
```diff
@@ -122,17 +122,17 @@ static void call_callback (void *key CTAGS_ATTR_UNUSED, void *value, void *user_
...
 
 	pushLanguage (client->lang);
 	client->callback (token, client->data);
 	popLanguage ();
 
 	if (token->type == YAML_STREAM_END_TOKEN)
 	{
-		if (client->teardownForTheLastInput)
-			client->teardownForTheLastInput (client->data);
+		if (client-> inputEnd)
```
Comment (b4n) [inline parsers/yaml.c]: extra space

File: parsers/ansibleplaybook.c [hunk 3]
```diff
@@ -185,10 +185,11 @@ static void ansiblePlaybook (yaml_token_t *token, void *data)
 
 static void *ansiblePlaybookPrepare(void)
 {
-	static struct ansiblePlaybookState state = {
-		.type_stack = NULL,
-		.play_detection_state = DSTAT_PLAY_NAME_INITIAL,
-	};
+	static struct ansiblePlaybookState state;
+
+	Assert (state.type_stack = NULL);
```
Comment (b4n) [inline parsers/ansibleplaybook.c]: is it really a good idea to perform such a check in the init phase?  If it's useful, shouldn't it rather be at the end of the previous run, as IIUC it checks whether the state was properly poped or something?

File: parsers/meta-m4.h (hunk 1 above)
Comment (masatake) [inline parsers/meta-m4.h]: `changequote()` is handled in `m4Quote{Open|Close}` in m4.c.
The const'ed fields are for initializing `m4Quote{Open|Close}`.

**Comment by masatake** [comment]
I found 861e49b is wrong. I will fix it.

File: parsers/meta-m4.h (hunk 1 above)
Comment (b4n) [inline parsers/meta-m4.h]: Oops, okay sorry.

File: parsers/ansibleplaybook.c [hunk 4]
```diff
@@ -181,13 +181,14 @@ static void ansiblePlaybook (yaml_token_t *token, void *data)
 		popBlockType (state, token);
 	else if (token->type == YAML_STREAM_END_TOKEN)
 		popAllBlockType (data, token);
+
+	Assert (state->type_stack = NULL);
```
Comment (b4n) [inline parsers/ansibleplaybook.c]: should be `==`, but I guess you noticed

File: parsers/pythonloggingconfig.c [hunk 5]
```diff
@@ -90,7 +90,7 @@ static bool pythonLoggingConfigProbeLanguage (const char *section, const char *k
 
 static struct iniconfParserClient PythonLoggingConfigIniconfClient = {
 	.lang = LANG_IGNORE,
-	.prepareForNewInput = pythonLoggingConfigPrepareForNewInput,
+	.inputStart = pythonLoggingConfiginputStart,
```
Comment (b4n) [inline parsers/pythonloggingconfig.c]: alignment

**Comment by b4n** [comment]
otherwise LGTM

**Pull request merged**